vt/emulate.go

```
@@ -1796,30 +1884,56 @@ func (em *emulator) putRune(r rune) {
-				// we are adding to a cluster
-				em.cells[lastIdx].C = cs
-				em.cells[lastIdx].W = width
-				col := Col(lastIdx) % dim.X
-				row := Row(lastIdx / int(dim.X))
-				// we may have to move position if this switches to wide, so recalculate expected end
-				end := (col + Col(width)) % dim.X
-				if em.getPrivateMode(PmAutoMargin) == ModeOn && end >= dim.X {
-					em.autoWrap = true
+			// ASCII-to-ASCII pairs cannot extend a grapheme cluster, except CRLF.
+			prev := em.cells[lastIdx].C
+			if len(prev) == 1 && prev[0] < utf8.RuneSelf && shouldCheckGrapheme(prev[0], r) == false {
```

golangci-lint (2.11.4)

[error] S1002: should omit comparison to bool constant, can be simplified to !shouldCheckGrapheme(prev[0], r)

(staticcheck)